src/StreamJsonRpc/JsonRpc.cs

```diff
@@ -184,6 +184,11 @@ public JsonRpc(IJsonRpcMessageHandler messageHandler, object target)
         /// </remarks>
         public JsonRpc(IJsonRpcMessageHandler messageHandler)
         {
+            if (messageHandler.Formatter is IJsonRpcInstanceContainer formatter)
+            {
+                formatter.Rpc = this;
+            }
+
             Requires.NotNull(messageHandler, nameof(messageHandler));
```

Comment:
This should go underneath the Requires.NotNull line. Otherwise you can throw NRE here instead of ANE.